### PR TITLE
ENH containers and bot version fetching task

### PR DESCRIPTION
**Pull request by beckermr**
This PR is another pass at getting key bot tasks running in containers.

todo

- [x] move different tasks to separate PRs and write tests (https://github.com/regro/cf-scripts/pull/2376)
- [x] move ruff to another PR (https://github.com/regro/cf-scripts/pull/2374)
- ~[x] hack on migrators in containers~ (deferred)
- [x] write docs for lazy json (https://github.com/regro/cf-scripts/pull/2375)
- [x] move lazy json changes to new PR (https://github.com/regro/cf-scripts/pull/2375)

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -38,3 +38,13 @@ line-length = 88
...
+
+[tool.ruff]
+extend-exclude = ["conda_forge_tick/migrators/disabled/legacy.py"]
+
+[tool.ruff.lint]
+select = ["E", "F", "W"]
+preview = true
+
+[tool.ruff.lint.pycodestyle]
+max-line-length = 120
```
Comment (beckermr) [inline pyproject.toml:50]:
```suggestion
```

File: tests/model/test_validate.py [hunk 2]
```diff
@@ -143,8 +143,10 @@ def pytest_generate_tests(metafunc):
     packages = get_all_feedstocks()
 
     if not packages:
-        raise ValueError(
-            "No packages found. Make sure the cf-graph is in the current working directory."
+        warnings.warn(
+            "No node attributes files found. Make sure these tests are run "
```
Comment (beckermr) [inline tests/model/test_validate.py:147]:
```suggestion
            "No packages found. Make sure these tests are run "
```

**Comment by beckermr** [comment]
@jaimergp here is a start on using a container for parsing versions. I am going to merge for now since I want to test performance before we do much more. You should feel free to leave review here in the PR though and I can fix things up later.

File: docker/install_cmds [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash -il
+
+set -exo pipefail
```
Comment (jaimergp) [inline docker/install_cmds:3]: `-euxo pipefail` wouldn't hurt either.

File: tests/test_container_tasks.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            "conda-forge-tick:test",
+            "python",
+            "/opt/autotick-bot/docker/run_bot_task.py",
+            "update-version",
+            "--existing-feedstock-node-attrs",
+            existing_feedstock_node_attrs,
+        ],
+        capture_output=True,
+    )
+    assert res.returncode == 0
```
Comment (jaimergp) [inline tests/test_container_tasks.py:58]: You can pass `check=True` to `subprocess.run` for the same check plus extra verbosity in the exception.

**Pull request merged**